static-analysis/autodispose-lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt

```diff
@@ -126,9 +128,13 @@ class AutoDisposeDetector: Detector(), SourceCodeScanner {
 
     if (isReactiveType(evaluator, method)
         && isInScope(evaluator, node.getContainingUClass())
-        && isExpressionValueUnused(node)
     ) {
-      context.report(ISSUE, node, context.getLocation(node), LINT_DESCRIPTION)
+      val isUnusedReturnValue = isExpressionValueUnused(node)
+      if (isUnusedReturnValue || !isCapturedTypeAllowed(node.returnType, method)) {
```

Comment:
Can you add a test for method references? Both in java and kotlin

Reply:
443573d7e5f85bcb60869d9aeec69f4c2c8eb5d4